### DISCUSSION TITLE
Outlook Office 365/Windows Mail

### DISCUSSION THREAD

**Discussion by renepaalhaar** [markuspoerschke/iCal#257] [Q&A]
Hi,

In Windows Mail I'm seeing the 'Accept, Tentative and Decline' buttons. When I set RSVP to False, I still see the buttons. Can I somehow manage these buttons, I just want an 'Add to Calendar'-button. Or do I have to dive into the Windows Mail to change these buttons?

My Outlook Office 365 is only showing the appointment details (subject, location, start time, end time). But there is no option/no button to add it to the Outlook Calendar. Is this something I have to manage in Outlook?

In short, can I manage my ics-attachment in Windows Mail/Outlook, to add/save it to the Windows Calendar/Outlook Calendar?

Hope to hear from you!

Rene

**Comment by markuspoerschke** [comment]
Can you provide a minimal code example? Also a generated `ics`-file might help.

**Comment by renepaalhaar** [comment]
Hello, 

Below you can find the code I use to generate the ics. 
    
    $vCalendar = new Calendar('-//XXX//EN');
    $vCalendar->setMethod(Calendar::METHOD_PUBLISH);

    $dstRecurrenceRule = new RecurrenceRule();
    $dstRecurrenceRule->setFreq(RecurrenceRule::FREQ_YEARLY)
        ->setByMonth(3)
        ->setByDay('-1SU');

    $vTimezoneRuleDst = new TimezoneRule(TimezoneRule::TYPE_DAYLIGHT);
    $vTimezoneRuleDst->setTzName('CEST')
        ->setDtStart(new \DateTime('1981-03-29 02:00:00', $dtz))
        ->setTzOffsetFrom('+0100')
        ->setTzOffsetTo('+0200')
        ->setRecurrenceRule($dstRecurrenceRule);

    $stdRecurrenceRule = new RecurrenceRule();
    $stdRecurrenceRule->setFreq(RecurrenceRule::FREQ_YEARLY)
        ->setByMonth(10)
        ->setByDay('-1SU');

    $vTimezoneRuleStd = new TimezoneRule(TimezoneRule::TYPE_STANDARD);
    $vTimezoneRuleStd->setTzName('CET')
        ->setDtStart(new \DateTime('1996-10-27 03:00:00', $dtz))
        ->setTzOffsetFrom('+0200')
        ->setTzOffsetTo('+0100')
        ->setRecurrenceRule($stdRecurrenceRule);

    $vTimezone = new Timezone($tz);
    $vTimezone->addComponent($vTimezoneRuleDst);
    $vTimezone->addComponent($vTimezoneRuleStd);
    $vCalendar->setTimezone($vTimezone);

    $vEvent = new Event();
    $vEvent->setUniqueId('xxx'.strtotime($plannedDateNew).$location['Location']['name'])
        ->setStatus(EVENT::STATUS_CONFIRMED)
        ->setTimeTransparency(EVENT::TIME_TRANSPARENCY_OPAQUE)
        ->setSequence(0)
        ->setDtStart($start)
        ->setDtEnd($end)
        ->setDescription(sprintf(__('Afspraak met %s'), $fullName))
        ->setSummary(sprintf(__('Afspraak bij %s'), $location['Location']['name']));

    if ($location && $address) {
        $vEvent->setLocation($address);
    }

    $vEvent->setUseTimezone(true);
   
    $vCalendar->addComponent($vEvent);
    $vIcs = $vCalendar->render();

And is there a way to make the location suitable for Outlook? In the screenshot below, you can see the address is in 1 rule (the zipcode is 4561 and house number 4), without comma or space. Should be something like Teststreet 111, 4567 XX, City. Can this be done?

![Screenshot 2020-03-03 at 15 39 34](https://user-images.githubusercontent.com/38661835/75787639-4cd0fa00-5d67-11ea-9c22-09f94690994b.png)

**Comment by steffjenl** [comment]
```
$address = 'Dorpstraat 11';
$zipcode = '1000 AA';
$city = 'Amsterdam';

if (!empty($address) && !empty($zipcode) && !empty($city)) {
    $vEvent->setLocation(sprintf('%s, %s %s', $address, $zipcode, $city));
}
```

will give location the value _Dorpstraat 11, 1000 AA Amsterdam_
